invoke-ai/InvokeAI
# Simple Installer for Unified Directory Structure, Initial Implementation

# A Smooth Install Experience (Maybe)

This PR implements the runtime directory layout scheme described in Discussion item #1789  and performs the install in such a way that everything, including the virtual environment and init file, are located in the `invokeai` runtime directory. It is dependent on Python 3.9 or 3.10 being installed on the target machine, but is not dependent on conda, git or any other package. (At startup the install script checks that the right version of Python is installed)

## Pros

- The install script logic is much simpler, and handles being stopped and restarted halfway through.
- Everything gets installed in one place, including the launch script, the init file and the `.venv` directory.
- The source code is never left around to confuse the user.
- Works on Linux CUDA and AMD
- Readline autocomplete support is preserved in CLI 

## Cons

- User has to install Python first.
- Not tested on Mac (please test)
- Windows .bat files still need to be written
- If user moves the `invokeai` directory everything breaks.

## To test:

1. Enter the directory `simple_installer`
2. Run `create_installers.sh` to create the three zip files. Or use the ones that are attached
[InvokeAI-simple-installer-2.2.4-linux.zip](https://github.com/invoke-ai/InvokeAI/files/10162019/InvokeAI-simple-installer-2.2.4-linux.zip)
[InvokeAI-simple-installer-2.2.4-mac.zip](https://github.com/invoke-ai/InvokeAI/files/10162020/InvokeAI-simple-installer-2.2.4-mac.zip)

3. Move `~/invokeai` out of the way if you plan to reuse this path.
4. Make sure to deactivate any existing conda/venv environment (but I'd be interested in knowing what happens if you don't deactivate them. The install script tries to inactivate any environments hanging around at startup, but I haven't tested if this works)
5. Go to your downloads/build directory.
6. unzip the .zip file for your platform
7. Execute `install.sh`
8. Go to your invokeai runtime directory and launch using `invoke.sh`
